### PR TITLE
Allow launch template spot instances without mixed policy

### PR DESCRIPTION
**Pull request by TarekAS**
# PR o'clock

## Description

Allow the creation of spot nodegroups using Launch Templates without resorting to Mixed Policy (due to certain incompatibilities with cluster autoscaler, for example). 

Edit: Switched to using `dynamic` block, instead of a separate `workers_launch_template_spot`.

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/*` directories
- [ ] CI tests are passing
- [x] I've added my change to CHANGELOG.md and highlighted any breaking changes
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by sppwf** [comment]
Hi @TarekAS 

Try next dynamic Block in the `workers_launch_template.tf` 
```hcl
  dynamic instance_market_options {
    iterator = item
    for_each = lookup(var.worker_groups_launch_template[count.index], "market_type", null) == null ? [] : list(lookup(var.worker_groups_launch_template[count.index], "market_type", null))
    content {
      market_type = item.value
    }
  }
````

Its quite ugly, however it will work. Conditional with for_each

Thanks

**Comment by TarekAS** [comment]
@sppwf I have already tried that as seen [here](https://github.com/terraform-aws-modules/terraform-aws-eks/pull/409#issuecomment-518141906). The issue is that it was causing new changes upon each terraform run.

Making its own option was the only working solution I found.

**Comment by sppwf** [comment]
@TarekAS i have used your code as well, and it does not cause new changes upon each run. 
U am using terragrunt, however it should not make any difference. 
I have used this default value: 
```hcl
locals {
  workers_group_defaults_defaults = {
    use_spot_instances   = false
  }
}
```
`null` value for the list passed to for_each should not trigger dynamic block to create `instance_market_options` block for launch_templates with  `use_spot_instances`   set to `false`

Thanks

**Comment by TarekAS** [comment]
@sppwf I see. Then it must have been something else from my side causing the change. I'll check again and confirm, if so I'll close this PR and create a new one with the better approach.

**Comment by TarekAS** [comment]
Looks like the dynamic block worked without causing new changes. I have updated this PR with the proper changes.

File: workers_launch_template.tf [hunk 1]
```diff
@@ -229,6 +229,14 @@ resource "aws_launch_template" "workers_launch_template" {
     )
   }
 
+  dynamic instance_market_options {
```
Comment (max-rocket-internet) [inline workers_launch_template.tf]: Awesome! Finally seeing some great use of the new features in TF 0.12. Elegant 💅

**Comment by max-rocket-internet** [comment]
Could we use the same method from this PR to remove `workers_launch_template_mixed.tf` and have the logic in `workers_launch_template.tf` without breaking anything?

**Pull request merged**